Assets/BossRoom/Scripts/Server/Game/Entity/IDamageable.cs

```diff
@@ -24,6 +25,19 @@ public interface IDamageable
         /// The transform of this object.
         /// </summary>
         Transform transform { get; }
+
+        //[Flags]
+        public enum SpecialDamageFlags
+        {
+            None                = 0,
+            StunBossOnImpact    = 1 << 0,
```

Comment:
Let's rename this to something more generic (StunnedOnImpact).

Reply:
ok. (Used `StunOnTrample`, to make it really clear what impact we're talking about)